tests/acceptance/shared/cluster.go

```diff
@@ -0,0 +1,283 @@
+	var res string
+	var err error
+	if destination == "host" {
+		res, err = RunCommandHost(cmd)
+		if err != nil {
+			return "", err
+		}
+	} else if destination == "node" {
+		ips := FetchNodeExternalIP()
+		for _, ip := range ips {
+			res, err = RunCommandOnNode(cmd, ip)
+			if err != nil {
+				return "", err
+			}
+		}
+	} else {
+		return "", fmt.Errorf("invalid destination: %s", destination)
+	}
+
+	return res, nil
+}
+
+// FetchClusterIP returns the cluster IP and port of the service.
+func FetchClusterIP(
+	namespace string,
+	serviceName string) (string, string, error) {
+	ip, err := RunCommandHost("kubectl get svc " + serviceName + " -n " + namespace +
+		" -o jsonpath='{.spec.clusterIP}' --kubeconfig=" + KubeConfigFile)
+	if err != nil {
+		return "", "", err
+	}
+
+	port, err := RunCommandHost("kubectl get svc " + serviceName + " -n " + namespace +
+		" -o jsonpath='{.spec.ports[0].port}' --kubeconfig=" + KubeConfigFile)
+	if err != nil {
+		return "", "", err
+	}
+
+	return ip, port, err
+}
+
+// FetchNodeExternalIP returns the external IP of the nodes.
+func FetchNodeExternalIP() []string {
+	res, _ := RunCommandHost("kubectl get nodes " +
+		"--output=jsonpath='{.items[*].status.addresses[?(@.type==\"ExternalIP\")].address}' " +
+		"--kubeconfig=" + KubeConfigFile)
+	nodeExternalIP := strings.Trim(res, " ")
+	nodeExternalIPs := strings.Split(nodeExternalIP, " ")
+
+	return nodeExternalIPs
+}
+
+// RestartCluster restarts the rke2 service on each node given by external IP.
+func RestartCluster(ip string) (string, error) {
+	return RunCommandOnNode("sudo systemctl restart rke2-*", ip)
+}
+
+// FetchIngressIP returns the ingress IP of the given namespace
+func FetchIngressIP(namespace string) ([]string, error) {
+	res, err := RunCommandHost(
+		"kubectl get ingress -n " +
+			namespace +
+			"  -o jsonpath='{.items[0].status.loadBalancer.ingress[*].ip}' --kubeconfig=" +
+			KubeConfigFile,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	ingressIP := strings.Trim(res, " ")
+	if ingressIP == "" {
+		return nil, nil
+	}
+	ingressIPs := strings.Split(ingressIP, " ")
+
+	return ingressIPs, nil
+}
+
+// Nodes returns nodes parsed from kubectl get nodes.
+func Nodes(print bool) ([]Node, error) {
+	nodes := make([]Node, 0, 10)
+
+	cmd := "kubectl get nodes --no-headers -o wide --kubeconfig=" + KubeConfigFile
+	res, err := RunCommandHost(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	nodelist := strings.TrimSpace(res)
+	split := strings.Split(nodelist, "\n")
+	for _, rec := range split {
+		if strings.TrimSpace(rec) != "" {
+			fields := strings.Fields(rec)
+			n := Node{
+				Name:       fields[0],
+				Status:     fields[1],
+				Roles:      fields[2],
+				Version:    fields[4],
+				InternalIP: fields[5],
+				ExternalIP: fields[6],
+			}
+			nodes = append(nodes, n)
+		}
+	}
+	if print {
+		fmt.Println(nodelist)
+	}
+
+	return nodes, nil
+}
+
+// Pods returns pods parsed from kubectl get pods.
+func Pods(print bool) ([]Pod, error) {
```

Comment:
Let's retain the name ParsePods as per its functionality

Reply:
sure done